src/ts/components/timelinemarkers.ts

```diff
@@ -0,0 +1,208 @@
+import { DOM } from '../dom';
+import { Component, ComponentConfig } from './component';
+import { TimelineMarker } from '../uiconfig';
+import { SeekBarMarker } from './seekbar';
+import { PlayerUtils } from '../playerutils';
+import { Timeout } from '../timeout';
+
+export interface MarkersConfig extends ComponentConfig {
+  /**
+   * Used for seekBar marker snapping range percentage
+   */
+  snappingRange?: number;
+}
+
+export class TimelineMarkers extends Component<MarkersConfig> {
+  private markersContainer: DOM;
+  private timelineMarkers: SeekBarMarker[];
+  private player: PlayerAPI;
+  private uimanager: UIInstanceManager;
+  private pausedTimeshiftUpdater: Timeout;
+  private getSeekBarWidth: () => number;
+  protected config: MarkersConfig;
+
+
+  constructor(config: MarkersConfig, getSeekBarWidth: () => number, markersContainer: DOM) {
+    super();
+    this.config = config;
+    this.getSeekBarWidth = getSeekBarWidth;
+    this.markersContainer = markersContainer;
+  }
+
+  public configure(player: PlayerAPI, uimanager: UIInstanceManager) {
+    this.player = player;
+    this.uimanager = uimanager;
+    this.configureMarkers();
+
+  }
+
+  private configureMarkers(): void {
+    // Remove markers when unloaded
+    this.player.on(this.player.exports.PlayerEvent.SourceUnloaded, () => this.clearMarkers());
+    // Update markers when the size of the seekbar changes
+    this.player.on(this.player.exports.PlayerEvent.PlayerResized, () => this.updateMarkersDOM());
+
+    this.player.on(this.player.exports.PlayerEvent.SourceLoaded, () => {
+      if (this.player.isLive()) {
+        // Update marker position as timeshift range changes
+        this.player.on(this.player.exports.PlayerEvent.TimeChanged, () => this.updateMarkers());
+        // Update marker postion when paused as timeshift range changes
+        this.configureLivePausedTimeshiftUpdater(() => this.updateMarkers());
+      }
+    });
+    this.uimanager.getConfig().events.onUpdated.subscribe(() => this.updateMarkers());
+    this.uimanager.onRelease.subscribe(() => this.uimanager.getConfig().events.onUpdated.unsubscribe(() => this.updateMarkers()));
+
+    // Init markers at startup
+    this.updateMarkers();
+  }
+
+  public getMarkerAtPosition(percentage: number): SeekBarMarker | null {
+    const snappingRange = this.config.snappingRange;
+
+    if (this.timelineMarkers.length > 0) {
+      for (let marker of this.timelineMarkers) {
+        // Handle interval markers
+        if (marker.duration > 0
+          && percentage >= marker.position - snappingRange
+          && percentage <= marker.position + marker.duration + snappingRange) {
+          return marker;
+        }
+        // Handle position markers
+        else if (percentage >= marker.position - snappingRange
+          && percentage <= marker.position + snappingRange) {
+          return marker;
+        }
+      }
+    }
+
+    return null;
+  }
+
+  private clearMarkers = (): void  => {
```

Comment:
Either do arrow functions or not everywhere.

Reply:
https://github.com/bitmovin/bitmovin-player-ui/pull/310/commits/61a805b60a52d8f5cfe13b0a69ae598ddb34abc1